components/shared/SearchBar.tsx

```diff
@@ -45,6 +48,7 @@ const SearchBar: React.FC = () => {
   const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     if (validAddress) {
+      setWalletAddress(address);
```

Comment:
Why do it here and not on the other page after the push?

Reply:
Actually I removed it already on my next commit. Kept this `setState` for login purpose.